### ISSUE TITLE
CYCLIC TERM for `=../2`

### ISSUE THREAD

**Issue by dram** [trealla-lang/trealla#191]
Following code will cause CYCLIC TERM error:

```
:- initialization(main).

foo(T) :- T =.. [bar, A].

main :-
        foo(T1),
        T2 =.. [baz, T1],
        write(T2), nl, halt.
```

Output:

```
% tpl --trace foo.pro
Call: ?- main
Call: main
Exit: main
Call: foo(T1)
Exit: foo(T)
Call: T=..[bar,A]
Exit: bar(A)=..[bar,A]
Call: T2=..[baz,bar(A)]
Exit: baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(ba...
Call: write(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(bar(baz(...
ERROR: ERROR ABORT_MAXDEPTH - CYCLIC TERM
```

**Comment by dram** [comment]
Fix confirmed, thanks.

**Comment by infradig** [comment]
Fixes #191 & #198. Now ?-Prolog crashes, but that is the lesser of two evils for now. I will look into it.
